python/packages/ai/teams/ai/ai.py

```diff
@@ -0,0 +1,103 @@
+"""
+Copyright (c) Microsoft Corporation. All rights reserved.
+Licensed under the MIT License.
+"""
+
+from logging import Logger
+from typing import Dict, Optional
+
+from teams.ai.action import ActionEntry, ActionFunction, DefaultActionTypes
+from teams.ai.exceptions import AIException
+
+
+class AI:
+    """
+    ### AI System
+
+    The AI system is responsible for generating plans, moderating input and output, and
+    generating prompts. It can be used free standing or routed to by the Application object.
+    """
+
+    _log: Logger
+    _actions: Dict[str, ActionEntry] = {}
+
+    def __init__(self, log=Logger("default")) -> None:
+        self._log = log
+        self._actions = {}
+        self.action(DefaultActionTypes.UNKNOWN_ACTION)(self.__unknown_action__)
+        self.action(DefaultActionTypes.FLAGGED_INPUT)(self.__flagged_input__)
+        self.action(DefaultActionTypes.FLAGGED_OUTPUT)(self.__flagged_output__)
+        self.action(DefaultActionTypes.RATE_LIMITED)(self.__rate_limited__)
+
+    def action(self, name: Optional[str] = None, allow_overrides=False):
```

Comment:
JS SDK requires user to provide a name, may also mark the `name` parameter as required?